### PR TITLE
Ignore any IDs generated by the DB

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

**Comment by tirsen** [comment]
This is causing us issues as we need to use MySQL directly and from Vitess for a transition period.

This could possibly be a command line flag but I think it's fine to just ignore the MySQL generated ID. That's after all what MySQL does, if you provide an ID when inserting it just ignored the autoinc column.

File: go/vt/vtgate/engine/route.go [hunk 1]
```diff
@@ -447,9 +447,6 @@ func (route *Route) execInsertUnsharded(vcursor VCursor, queryConstruct *queryin
 		return nil, fmt.Errorf("execInsertUnsharded: %v", err)
 	}
 	if insertid != 0 {
```
Comment (sougou) [inline go/vt/vtgate/engine/route.go]: You can remove this check.

File: go/vt/vtgate/router_dml_test.go [hunk 2]
```diff
@@ -722,6 +722,10 @@ func TestInsertFail(t *testing.T) {
 	if err == nil || !strings.HasPrefix(err.Error(), want) {
 		t.Errorf("routerExec: %v, want prefix %v", err, want)
 	}
+}
+
+func TestIgnoresAutoincGeneratedByMySQL(t *testing.T) {
```
Comment (sougou) [inline go/vt/vtgate/router_dml_test.go]: You can also delete this test. It's essentially obsolete now.

File: go/vt/vtgate/engine/route.go (hunk 1 above)
Comment (tirsen) [inline go/vt/vtgate/engine/route.go]: Done!

File: go/vt/vtgate/router_dml_test.go (hunk 2 above)
Comment (tirsen) [inline go/vt/vtgate/router_dml_test.go]: Done!

**Comment by tirsen** [comment]
Alright, comments addressed please have another look.

**Comment by sougou** [comment]
LGTM

**Pull request merged**